source/tyr/tyr/tasks.py

```diff
@@ -406,3 +407,19 @@ def remove_autocomplete_depot(name):
             logging.warn('no autocomplete directory for {}, removing nothing'.format(autocomplete_dir))
     else:
         logging.warn('no main autocomplete directory, removing nothing')
+
+@celery.task()
+def purge_autocomplete():
+    logger = logging.getLogger(__name__)
+    autocomplete_instances = models.AutocompleteParameter.query.all()
+    for ac_instance in autocomplete_instances:
+        logger.info('purging autocomplete backup directories for %s', ac_instance.name)
+        loaded = set(os.path.realpath(os.path.dirname(dataset.name))
+                 for dataset in ac_instance.last_datasets(5))
+        for directory in loaded:
+            if os.path.exists(directory):
```

Comment:
hum strange, you delete the last dataset ? isn't those the one you need to keep ?